includes/Admin/Cross_Origin_Isolation.php

```diff
@@ -118,40 +101,36 @@ public function admin_footer() {
 	 *
 	 * @param string $html HTML document as string.
 	 *
-	 * @return string
+	 * @return string Processed HTML document.
 	 */
 	protected function replace_in_dom( string $html ): string {
-		$document = Document::fromHtml( $html );
-
-		if ( ! $document ) {
-			return $html;
-		}
-
-		$map = [
-			'link'   => 'href',
-			'img'    => 'src',
-			'iframe' => 'src',
-			'script' => 'src',
+		$tags = [
+			'link',
+			'img',
+			'iframe',
+			'script',
 		];
 
+		$tags      = implode( '|', $tags );
+		$matches   = [];
 		$processed = [];
 
-		foreach ( $map as $tag => $attribute ) {
-			$tags = $document->getElementsByTagName( $tag );
-			foreach ( $tags as $node ) {
-				$value = $node->getAttribute( $attribute );
-				if ( ! $value ) {
+		if ( preg_match_all( '#<(?P<tag>' . $tags . ')[^<]*?(?:>[\s\S]*?</(?P=tag)>|\s*/>)#', $html, $matches ) ) {
```

Comment:
preg matches were avoid by design, we shouldn't implement there here.

Reply:
I know. Let's continue discussion at https://github.com/google/web-stories-wp/pull/8387#discussion_r672989172